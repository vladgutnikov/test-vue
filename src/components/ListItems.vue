<template>
  <v-container>
    <v-row class="justify-center">
      <v-col md="4">
        <v-card max-width="100%" class="mx-auto" v-show="!!items.length">
          <v-list three-line>
            <ListItem
              v-for="(item, index) in items"
              :key="`${index} ${item.header} `"
              :item="item"
            />
          </v-list>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
  import { VCard, VList, VCol, VRow, VContainer } from 'vuetify/lib';
  import ListItem from './ListItem.vue';

  export default {
    name: 'ListItems',
    components: {
      VCard,
      VList,
      VCol,
      VRow,
      VContainer,
      ListItem
    },
    props: {
      items: {
        type: Array,
        default: () => []
      }
    }
  };
</script>
