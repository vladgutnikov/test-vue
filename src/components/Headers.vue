<template>
  <div>
    <v-bottom-navigation shift height="auto">
      <v-container>
        <v-row>
          <v-col md="4" offset-md="4">
            <div class="wrapper-search">
              <v-text-field
                class="my-search"
                v-model="search"
                label="Search"
                outlined
                @input="getUserData"
                @blur="clearDataArray"
                prepend-inner-icon="search"
              ></v-text-field>
            </div>
          </v-col>
          <v-col md="4" class="d-flex justify-end">
            <v-btn>
              home
              <v-icon x-large dark>home</v-icon>
            </v-btn>
            <v-btn>
              My network
              <v-icon x-large dark>view_list</v-icon>
            </v-btn>
            <v-btn>
              Jobs
              <v-icon x-large dark>work</v-icon>
            </v-btn>
          </v-col>
        </v-row>
      </v-container>
    </v-bottom-navigation>
    <ListItems :items="filterData" />
  </div>
</template>

<script>
import ListItems from './ListItems';
import {
  VBtn,
  VIcon,
  VCol,
  VRow,
  VContainer,
  VBottomNavigation,
  VTextField,
} from 'vuetify/lib';
import data from '../data/data.json';

export default {
  data: () => ({ search: null, items: [], filterData: [] }),
  components: {
    ListItems,
    VBtn,
    VIcon,
    VCol,
    VRow,
    VContainer,
    VBottomNavigation,
    VTextField,
  },
  methods: {
    getUserData() {
      this.filterData = this.items.filter((item) => {
        if (item.title.toUpperCase().indexOf(this.search.toUpperCase()) > -1)
          return item;
      });
      if (this.filterData) this.sortData();
    },
    sortData() {
      this.filterData.sort((a, b) => {
        if (a.title.toLowerCase() < b.title.toLowerCase()) return -1;
        else return 1;
      });
    },
    clearDataArray() {
      if (!this.search) this.filterData = [];
    },
  },
  mounted() {
    this.items = [...data];
  },
};
</script>

<style>
.my-search {
  background: #ccc;
  height: 56px;
}
</style>
