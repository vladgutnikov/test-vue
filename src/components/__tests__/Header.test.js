import { createLocalVue, mount, shallowMount } from '@vue/test-utils';
import Headers from '../Headers.vue';
import ListItems from '../ListItems.vue';
import Vuetify from 'vuetify';
import Vue from 'vue';

describe('Home.vue', () => {
  let wrapper, vuetify, newWrapper;
  const localVue = createLocalVue();

  beforeEach(() => {
    Vue.use(Vuetify);
    wrapper = shallowMount(Headers);
    vuetify = new Vuetify();
    newWrapper = mount(Headers, {
      localVue,
      vuetify,
    });
  });

  it('renders a vue instance', () => {
    expect(shallowMount(Headers)).toBeTruthy();
  });

  it('check if child ListItems exists', () => {
    const child = wrapper.findComponent(ListItems);
    expect(child.exists()).toBe(true);
  });

  it('Enter text and check the value of inputText', () => {
    const textInput = newWrapper.find('.v-text-field input');
    textInput.setValue('84102');
    expect(newWrapper.vm.search).toBe('84102');
  });

  it('component snapshot Home.vue', () => {
    expect(newWrapper.html()).toMatchSnapshot();
  });

  it('set data', () => {
    newWrapper.setData({ search: 'bar' });
    expect(newWrapper.vm.search).toBe('bar');
  });
});
