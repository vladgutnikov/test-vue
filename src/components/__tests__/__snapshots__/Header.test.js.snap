// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue should have match snapshot 1`] = `
"<div>
  <div class=\\"v-bottom-navigation v-item-group theme--light v-bottom-navigation--shift\\" style=\\"height: auto; transform: none;\\">
    <div class=\\"container\\">
      <div class=\\"row\\">
        <div class=\\"col-md-4 offset-md-4 col\\">
          <div class=\\"wrapper-search\\">
            <div class=\\"v-input my-search theme--light v-text-field v-text-field--enclosed v-text-field--outlined\\">
              <div class=\\"v-input__control\\">
                <div class=\\"v-input__slot\\">
                  <div class=\\"v-input__prepend-inner\\">
                    <div class=\\"v-input__icon v-input__icon--prepend-inner\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate material-icons theme--light\\">search</i></div>
                  </div>
                  <fieldset aria-hidden=\\"true\\">
                    <legend style=\\"width: 0px;\\"><span>â€‹</span></legend>
                  </fieldset>
                  <div class=\\"v-text-field__slot\\"><label for=\\"input-95\\" class=\\"v-label theme--light\\" style=\\"left: 0px; position: absolute;\\">Search</label><input id=\\"input-95\\" type=\\"text\\"></div>
                </div>
                <div class=\\"v-text-field__details\\">
                  <div class=\\"v-messages theme--light\\">
                    <transition-group-stub tag=\\"div\\" name=\\"message-transition\\" class=\\"v-messages__wrapper\\"></transition-group-stub>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div class=\\"d-flex justify-end col-md-4 col\\"><button type=\\"button\\" class=\\"v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default\\"><span class=\\"v-btn__content\\">
            home
            <i aria-hidden=\\"true\\" class=\\"v-icon notranslate material-icons theme--dark\\" style=\\"font-size: 40px;\\">home</i></span></button> <button type=\\"button\\" class=\\"v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default\\"><span class=\\"v-btn__content\\">
            My network
            <i aria-hidden=\\"true\\" class=\\"v-icon notranslate material-icons theme--dark\\" style=\\"font-size: 40px;\\">view_list</i></span></button> <button type=\\"button\\" class=\\"v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default\\"><span class=\\"v-btn__content\\">
            Jobs
            <i aria-hidden=\\"true\\" class=\\"v-icon notranslate material-icons theme--dark\\" style=\\"font-size: 40px;\\">work</i></span></button></div>
      </div>
    </div>
  </div>
  <div class=\\"container\\">
    <div class=\\"row justify-center\\">
      <div class=\\"col-md-4 col\\">
        <div class=\\"mx-auto v-card v-sheet theme--light\\" style=\\"max-width: 100%; display: none;\\">
          <div role=\\"list\\" class=\\"v-list v-sheet theme--light v-list--three-line\\"></div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
